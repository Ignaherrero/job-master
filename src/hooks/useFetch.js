/* eslint-disable no-debugger */
/* eslint-disable no-unused-vars */
import { useEffect, useState } from "react";

import { geoFindMe } from "../utils/getLocation";

const useFetch = (jobName, cityNameInput, fullTime) => {
  const [state, setState] = useState({
    data: [],
    isLoading: true,
  });

  const latitude = 0;
  const longitude = 0;

  useEffect(() => {
    fetch(
      `https://jobs.github.com/positions.json?description=${jobName}&page=0&full_time=${fullTime}&${
        longitude > 0
          ? `long=${longitude}&lat=${latitude}`
          : `&location=${cityNameInput}`
      }`
    )
      .then((res) => res.json())
      .then((data) => {
        setState({ data, isLoading: false });
      })
      .catch((e) => console.log(e));
  }, [cityNameInput, jobName, fullTime]);

  return state;
};

export { useFetch };
