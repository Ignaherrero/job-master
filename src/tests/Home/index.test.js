import React from "react";

import { useFetch } from "../../hooks/useFetch.js";
import { Home } from "../../pages/Home/index.jsx";

const { shallow } = require("enzyme");

jest.mock("../../hooks/useFetch");

describe("<Home/>", () => {
  test("Debe renderizar el componente", () => {
    useFetch.mockReturnValue({
      data: [],
    });
    const wrapper = shallow(<Home />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Debe renderizar lo que devuelva el fetch", () => {
    useFetch.mockReturnValue({
      data: [
        {
          id: "0691096c-1b44-47e7-9767-15fe5e1dec5a",
          type: "Full Time",
          url: "https://jobs.github.com/positions/0691096c-1b44-47e7-9767-15fe5e1dec5a",
          created_at: "Fri May 07 17:56:37 UTC 2021",
          company: "ISE",
          company_url: "http://ISE.io",
          location: "Baltimore, MD",
          title: "Mid-Senior Full Stack Developer",
          description: " @ISEsecurity and @IoTvillage",
          how_to_apply:
            '<p>Apply at <a href="https://www.ise.io/careers/">https://www.ise.io/careers/</a></p>\n',
          company_logo:
            "https://jobs.github.com/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBcmlpIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--51899c4af7f26c0b14708f451979b9a803096b39/ISE_Logo_rgb.jpg",
        },
      ],
    });

    const wrapper = shallow(<Home />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Debe cambiar fulltime a false", () => {
    const wrapper = shallow(<Home />);

    console.log(wrapper.text().include("Full Time").simulate("clic"));
  });

  test("Debe retornar la propiedad fulltime true", () => {});
});
