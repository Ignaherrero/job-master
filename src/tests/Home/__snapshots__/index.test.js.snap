// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> Debe renderizar el componente 1`] = `
<Fragment>
  <styled.div>
    <styled.h1>
      Github 
      <span>
        Jobs
      </span>
    </styled.h1>
    <Form>
      <div>
        <div>
          <div>
            <label
              htmlFor=""
            >
              <Input
                handleChange={[Function]}
                name="jobName"
                placeholder="title, company, expertice"
                type="text"
              />
            </label>
            <Input
              type="submit"
              value="Search"
            />
          </div>
        </div>
      </div>
      <label
        onChange={[Function]}
      >
        <span>
          <ForwardRef(SvgCheck) />
        </span>
        Full Time
        <Input
          dontShowPromptCheckbox={true}
          name="fullTime"
          type="checkbox"
          value={true}
        />
      </label>
      <span>
        Location
      </span>
      <label
        htmlFor=""
        onChange={[Function]}
      >
        <Input
          name="cityNameInput"
          placeholder="City, state, zip code or country"
          type="text"
        />
      </label>
      <fieldset>
        <label
          htmlFor="london"
          onChange={[Function]}
        >
          <Input
            checked={false}
            id="london"
            name="cityNameInput"
            type="radio"
            value="london"
          />
          London
        </label>
        <label
          htmlFor="amsterdam"
          onChange={[Function]}
        >
          <Input
            checked={false}
            id="amsterdam"
            name="cityNameInput"
            type="radio"
            value="amsterdam"
          />
          Amsterdam
        </label>
        <label
          htmlFor="newyork"
          onChange={[Function]}
        >
          <Input
            checked={false}
            id="new york"
            name="cityNameInput"
            type="radio"
            value="new york"
          />
          New York
        </label>
        <label
          htmlFor="berlin"
          onChange={[Function]}
        >
          <Input
            checked={false}
            id="berlin"
            name="cityNameInput"
            type="radio"
            value="berlin"
          />
          Berlin
        </label>
      </fieldset>
    </Form>
    <h1>
      Ningun job en tu ubicaci√≥n
    </h1>
    <s
      paginate={[Function]}
      postsPerPage={5}
      totalPosts={0}
    />
    <Footer />
  </styled.div>
</Fragment>
`;

exports[`<Home/> Debe renderizar lo que devuelva el fetch 1`] = `
<Fragment>
  <styled.div>
    <styled.h1>
      Github 
      <span>
        Jobs
      </span>
    </styled.h1>
    <Form>
      <div>
        <div>
          <div>
            <label
              htmlFor=""
            >
              <Input
                handleChange={[Function]}
                name="jobName"
                placeholder="title, company, expertice"
                type="text"
              />
            </label>
            <Input
              type="submit"
              value="Search"
            />
          </div>
        </div>
      </div>
      <label
        onChange={[Function]}
      >
        <span>
          <ForwardRef(SvgCheck) />
        </span>
        Full Time
        <Input
          dontShowPromptCheckbox={true}
          name="fullTime"
          type="checkbox"
          value={true}
        />
      </label>
      <span>
        Location
      </span>
      <label
        htmlFor=""
        onChange={[Function]}
      >
        <Input
          name="cityNameInput"
          placeholder="City, state, zip code or country"
          type="text"
        />
      </label>
      <fieldset>
        <label
          htmlFor="london"
          onChange={[Function]}
        >
          <Input
            checked={false}
            id="london"
            name="cityNameInput"
            type="radio"
            value="london"
          />
          London
        </label>
        <label
          htmlFor="amsterdam"
          onChange={[Function]}
        >
          <Input
            checked={false}
            id="amsterdam"
            name="cityNameInput"
            type="radio"
            value="amsterdam"
          />
          Amsterdam
        </label>
        <label
          htmlFor="newyork"
          onChange={[Function]}
        >
          <Input
            checked={false}
            id="new york"
            name="cityNameInput"
            type="radio"
            value="new york"
          />
          New York
        </label>
        <label
          htmlFor="berlin"
          onChange={[Function]}
        >
          <Input
            checked={false}
            id="berlin"
            name="cityNameInput"
            type="radio"
            value="berlin"
          />
          Berlin
        </label>
      </fieldset>
    </Form>
    <Card
      company="ISE"
      companyLogo="https://jobs.github.com/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBcmlpIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--51899c4af7f26c0b14708f451979b9a803096b39/ISE_Logo_rgb.jpg"
      createdAt="Fri May 07 17:56:37 UTC 2021"
      description=" @ISEsecurity and @IoTvillage"
      key="0691096c-1b44-47e7-9767-15fe5e1dec5a"
      location="Baltimore, MD"
      title="Mid-Senior Full Stack Developer"
      type="Full Time"
    />
    <s
      paginate={[Function]}
      postsPerPage={5}
      totalPosts={1}
    />
    <Footer />
  </styled.div>
</Fragment>
`;
